frontend/test/metabase/scenarios/dashboard-filters/parameters.cy.spec.js

```diff
@@ -198,59 +199,101 @@ describe("scenarios > dashboard > parameters", () => {
-    visitDashboard(1);
-    cy.findByTextEnsureVisible("Created At");
+    const questionDetails = {
+      query: {
+        "source-table": ORDERS_ID,
+        limit: 5,
+      },
+    };
 
-    // Add filter and save dashboard
-    cy.icon("pencil").click();
-    cy.icon("filter").click();
-    cy.findByText("Text or Category").click();
-    cy.findByText("Ends with").click();
+    const filter = {
+      name: "Text ends with",
+      slug: "text_ends_with",
+      id: "88a1257c",
+      type: "string/ends-with",
+      sectionId: "string",
+    };
 
-    // map the parameter to the Category field
-    selectFilter(cy.get(".DashCard"), "Category");
+    const dashboardDetails = {
+      parameters: [filter],
+    };
 
-    cy.findByText("Save").click();
+    cy.createQuestionAndDashboard({ questionDetails, dashboardDetails }).then(
+      ({ body: { id, card_id, dashboard_id } }) => {
+        cy.request("PUT", `/api/dashboard/${dashboard_id}/cards`, {
+          cards: [
+            {
+              id,
+              card_id,
+              row: 0,
+              col: 0,
+              sizeX: 12,
+              sizeY: 8,
+              series: [],
+              visualization_settings: {},
+              parameter_mappings: [
+                {
+                  parameter_id: filter.id,
+                  card_id,
+                  target: [
+                    "dimension",
+                    [
+                      "field",
+                      PRODUCTS.CATEGORY,
+                      {
+                        "source-field": ORDERS.PRODUCT_ID,
+                      },
+                    ],
+                  ],
+                },
+              ],
+            },
+          ],
+        });
 
-    // wait for saving to finish
-    cy.contains("You're editing this dashboard.").should("not.exist");
+        visitDashboard(dashboard_id);
+        cy.findByTextEnsureVisible("Created At");
+      },
+    );
 
     // populate the filter input
-    cy.findByText("Text ends with").click();
-    popover()
-      .find("input")
-      .type("zmo");
-
-    popover()
-      .contains("Add filter")
-      .click();
+    filterWidget().click();
+    cy.findByPlaceholderText("Enter some text").type("zmo{enter}");
+    cy.button("Add filter").click();
 
     cy.log(
       "**URL is updated correctly with the given parameter at this point**",
     );
-    cy.url().should("include", "text_ends_with=zmo");
+    cy.location("search").should("eq", "?text_ends_with=zmo");
 
     // Remove filter name
     cy.icon("pencil").click();
     cy.get(".Dashboard")
       .find(".Icon-gear")
       .click();
+
     cy.findByDisplayValue("Text ends with")
-      .click()
-      .clear();
-    cy.findByText("Save").click();
-    cy.wait("@dashboard");
-    cy.findByText("You're editing this dashboard.").should("not.exist");
+      .clear()
+      .blur();
+
+    cy.findByDisplayValue("unnamed");
+
+    cy.location("search").should("eq", "?unnamed=zmo");
```

Comment:
What is happening here? Specifically where is `unnamed` coming from?

Reply:
When you remove filter name and save, it automatically renames it to "unnamed".

Reply:
That is reflected in the URL just prior to save. Before this issue was fixed, the url didn't remove the reference even after you save dashboard.

Reply:
Yeah, I know that, it was to prevent empty filter names, which could completely bork a dashboard.
It would just have been nice with a little comment describing what is going on.